### PR TITLE
[MXNET-1213] add Cent OS build for Scala

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Add Cent OS build for Scala package
@marcoabreu @nswamy @andrewfayres @zachgk @piyushghai @yzhliu 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by lanking520** [comment]
Never saw this kind of undefined symbol, but might relate to this:
https://github.com/davisking/dlib/issues/1230
looks like is `libgcc` problem.

File: ci/docker/install/ubuntu_scala.sh [hunk 1]
```diff
@@ -29,12 +29,6 @@ apt-get update
 apt-get install -y openjdk-8-jdk
 apt-get install -y openjdk-8-jre
 
-echo "deb https://dl.bintray.com/sbt/debian /" | tee -a /etc/apt/sources.list.d/sbt.list
```
Comment (marcoabreu) [inline ci/docker/install/ubuntu_scala.sh]: Could you elaborate?

Comment (lanking520) [reply ci/docker/install/ubuntu_scala.sh]: The reason to remove sbt and Scala is we don't need these components to run a maven project. There is Scala compiler embedded in the maven project. In that case, sbt and Scala installation seemed to be unnecessary

File: Jenkinsfile [hunk 2]
```diff
@@ -698,6 +698,17 @@ core_logic: {
         }
       }
     },
+    'Scala: CentOS CPU': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/build-centos7-cpu') {
```
Comment (zachgk) [inline Jenkinsfile]: Change the ws to something unique (ut-scala-centos7-cpu) and change utils.unpack_and_init from from the ubuntu based 'cpu' to 'centos7_cpu'

File: Jenkinsfile [hunk 3]
```diff
@@ -698,6 +698,17 @@ core_logic: {
         }
       }
     },
+    'Scala: CentOS CPU': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/ut-scala-centos7-cpu') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            utils.unpack_and_init('centos7_cpu', mx_lib, true)
```
Comment (zachgk) [inline Jenkinsfile]: mx_lib should be mx_dist_lib. Also, change mx_lib to mx_dist_lib on line 155 as well. mx_dist_lib packs and unpacks more files which are necessary for scala build.

Comment (lanking520) [reply Jenkinsfile]: What diff on the files that mx_dist_lib will provide?

Comment (zachgk) [reply Jenkinsfile]: It adds 3rdparty/ps-lite/build/libps.a, deps/lib/libprotobuf-lite.a, and deps/lib/libzmq.a. According to the comment at the top, they are necessary for scala with dist_kvstore

**Comment by kalyc** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: ci/docker/install/centos7_scala.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# under the License.
+
+# build and install are separated so changes to build don't invalidate
+# the whole docker cache for the image
+
+set -ex
+
+yum install -y java-1.8.0-openjdk-devel
+# Build from source with Maven
+wget http://www.eu.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz
```
Comment (nswamy) [inline ci/docker/install/centos7_scala.sh]: can you uninstall existing version of maven `yum remove -y maven`

Comment (lanking520) [reply ci/docker/install/centos7_scala.sh]: There is no maven built in Cent OS 7.

**Comment by kalyc** [comment]
@mxnet-label-bot update [pr-awaiting-merge]

**Pull request merged**